### PR TITLE
Use multiple branches in CI to fix PR to another feature branch

### LINKED ISSUE #591: CI PR to another feature branch
When the target branch of a PR is another feature branch, so not the default branch, it should checkout all other packages to that branch as well. That is not happening right now.

### LINKED ISSUE #643: Targets not checked-out to master, when feature branch is removed  on GH since docker created
When a feature branch is available in the targets repo at moment of docker creation. The targets are checked-out to that branch. When the docker is used. It is possible the feature branch of the targets repo is already deleted/merged. The current master should be used in that case. But this isn't the case right now.

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
Review and merge after #563

Fixes #591
Fixes #643 

Do not merge: Remove `--debug`

Test run of PR to another feature branch: https://github.com/tue-robotics/rgbd/actions/runs/4092072454/jobs/7065877146
Test run of PR to default branch: https://github.com/tue-robotics/rgbd/actions/runs/4097481327/jobs/7065889606

### CODE REVIEW

**Review by ar13pit — APPROVED** [review]

**Comment by ar13pit** [comment]
Only approving the changes of this PR, not of the base branch :)

**Comment by MatthijsBurgh** [comment]
Switching back to default branch behaviour:
```
[tue-env-targets] Updating targets... From https://github.com/tue-robotics/tue-env-targets
 - [deleted]         (none)     -> origin/blaat
Your configuration specifies to merge with the ref 'refs/heads/blaat'
from the remote, but no such ref was fetched.
[tue-env-targets] Current branch 'blaat' isn't available anymore, switching to the default branch
Switched to branch 'master'
Your branch is behind 'origin/master' by 7 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
Branch 'master' set up to track remote branch 'master' from 'origin'.
Updating ab78d04..efd6b39
Fast-forward
 .github/workflows/main.yml | 26 +++++++++++++++++++++-----
 README.md                  |  5 +++++
 libboost-dev/install.yaml  |  2 ++
 ros/install.bash           |  4 +++-
 4 files changed, 31 insertions(+), 6 deletions(-)
 create mode 100644 README.md
 create mode 100644 libboost-dev/install.yaml
Synchronizing submodule url for 'hsr-setup'
Cloning into '/home/docker/ros/noetic/.env/targets/hsr-setup'...
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:tue-robotics/hsr-setup.git' into submodule path '/home/docker/ros/noetic/.env/targets/hsr-setup' failed
Failed to clone 'hsr-setup'. Retry scheduled
Cloning into '/home/docker/ros/noetic/.env/targets/hsr-setup'...
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:tue-robotics/hsr-setup.git' into submodule path '/home/docker/ros/noetic/.env/targets/hsr-setup' failed
Failed to clone 'hsr-setup' a second time, aborting
[tue-env-targets] Switched to the default branch, 'master'
[tue-env-targets] Trying to switch to branch 'PULLREQUEST'...
[tue-env-targets] Branch 'PULLREQUEST' does not exist. Current branch is 'master'
[tue-env-targets] Trying to switch to branch 'blaat'...
[tue-env-targets] Branch 'blaat' does not exist. Current branch is 'master'
```
(I manually forced master to be behind)

File: ci/install-package.sh [hunk 1]
```diff
@@ -222,17 +222,18 @@ then
     docker exec -t tue-env bash -c 'source ~/.bashrc; git -C "$TUE_SYSTEM_DIR"/src/"$PACKAGE" fetch origin "$REF_NAME"/"$PULL_REQUEST"/merge:PULLREQUEST'
 
     # Install the package completely
-    echo -e "\e[35m\e[1mtue-get install ros-$PACKAGE --test-depend --branch=PULLREQUEST\e[0m"
-    docker exec tue-env bash -c 'source ~/.bashrc; tue-get install ros-"$PACKAGE" --test-depend --branch=PULLREQUEST'
+    branch_string=${BRANCH:+" --try-branch=${BRANCH}"}
+    echo -e "\e[35m\e[1mtue-get install ros-$PACKAGE --test-depend${branch_string} --try-branch=PULLREQUEST\e[0m"
+    docker exec tue-env bash -c 'source ~/.bashrc; tue-get install ros-"${PACKAGE}" --test-depend --try-branch="${BRANCH}" --try-branch=PULLREQUEST'
```
Comment (ar13pit) [inline ci/install-package.sh:227]: I think we should use the branch_string here as well

File: ci/install-package.sh [hunk 2]
```diff
@@ -222,17 +222,18 @@ then
...
     # Checkout -f to be really sure
     echo -e "\e[35m\e[1mgit -C ~${TUE_SYSTEM_DIR#"${DOCKER_HOME}"}/src/$PACKAGE checkout -f PULLREQUEST --\e[0m"
     docker exec -t tue-env bash -c 'source ~/.bashrc; git -C "$TUE_SYSTEM_DIR"/src/"$PACKAGE" checkout -f PULLREQUEST --'
 else
     # Install the package
-    branch_string="${BRANCH:+" --branch=${BRANCH}"}"
+    branch_string=${BRANCH:+" --try-branch=${BRANCH}"}
     echo -e "\e[35m\e[1mtue-get install ros-${PACKAGE} --test-depend${branch_string}\e[0m"
-    docker exec tue-env bash -c 'source ~/.bashrc; tue-get install ros-"${PACKAGE}" --test-depend --branch="${BRANCH}"'
+    docker exec tue-env bash -c 'source ~/.bashrc; tue-get install ros-"${PACKAGE}" --test-depend --try-branch="${BRANCH}"'
```
Comment (ar13pit) [inline ci/install-package.sh:236]: Similarly here too

**Review by ar13pit — CHANGES_REQUESTED** [review]

File: ci/install-package.sh (hunk 1 above)
Comment (MatthijsBurgh) [reply ci/install-package.sh:227]: We don't do this later in the script. Which you also commented. It doesn't matter when  the argument is empty. But is doesn't hurt to change it either

Comment (ar13pit) [reply ci/install-package.sh:227]: Just for the sake for uniformity otherwise it only becomes a fancy print statement :)

Comment (MatthijsBurgh) [reply ci/install-package.sh:227]: If we just drop in the `branch_string` variable? Will it still be resolved inside the docker?

Comment (ar13pit) [reply ci/install-package.sh:227]: I hope so :laughing:

**Review by ar13pit — APPROVED** [review]

**Review by ar13pit — APPROVED** [review]

**Pull request merged**